src/Compilers/Core/Portable/InternalUtilities/SetWithInsertionOrder.cs

```diff
@@ -88,6 +96,7 @@ public IEnumerator<T> GetEnumerator()
 
         public ImmutableArray<T> AsImmutable() => _elements.ToImmutableArrayOrEmpty();
 
-        public T this[int i] => _elements[i];
+        [MaybeNull]
+        public T this[int i] => _elements is null ? default : _elements[i];
```

Comment:
>_elements is null ? default : _elements[i] [](start = 32, length = 42)

Callers shouldn't use `this[int i]` if there are no elements in the set.

Reply:
If you think it's worth fixing I can file a follow up issue but want to keep annotation separate from behavior changes as much as possible.

---
In reply to: [356899343](https://github.com/dotnet/roslyn/pull/40268#discussion_r356899343) [](ancestors = 356899343)

Reply:
It looks like the original code needs `!` only: `=> _elements![i];`.

---
In reply to: [357433119](https://github.com/dotnet/roslyn/pull/40268#discussion_r357433119) [](ancestors = 357433119,356899343)